### PR TITLE
Use a namedtuple for Camera Resolution, not a dict

### PR DESCRIPTION
**Pull request by chriselion**
The string keys were bothering me, and it now encapsulates some of the greyscale/channels logic.

### CODE REVIEW

File: ml-agents-envs/mlagents/envs/brain.py [hunk 1]
```diff
@@ -5,19 +5,33 @@
...
 
 
+class CameraResolution(NamedTuple):
+    height: int
+    width: int
+    gray_scale: bool
+
+    @property
+    def num_channels(self) -> int:
+        return 1 if self.gray_scale else 3
```
Comment (ervteng) [inline ml-agents-envs/mlagents/envs/brain.py]: I almost want to say CameraResolution shouldn't be aware of whether it's grey scale or not, just how many channels it has. Seems more generalizable that way. We could pass it in 1 or 3 on creation from the proto instead - what do you think?

Comment (chriselion) [reply ml-agents-envs/mlagents/envs/brain.py]: Agreed, but that would be a much bigger change (would require communicator proto changes), and will (hopefully) come for free with the sensor refactor.

Comment (ervteng) [reply ml-agents-envs/mlagents/envs/brain.py]: OK, if it's coming anyways in the sensor refactor, no sense in doing it twice.

Comment (chriselion) [reply ml-agents-envs/mlagents/envs/brain.py]: Yeah, should be coming then, and also gives me some more confidence while doing the refactoring.

File: gym-unity/gym_unity/envs/__init__.py [hunk 2]
```diff
@@ -132,18 +132,14 @@ def __init__(
...
-                depth = 3
             self._observation_space = spaces.Box(
                 0,
                 1,
                 dtype=np.float32,
                 shape=(
-                    brain.camera_resolutions[0]["height"],
-                    brain.camera_resolutions[0]["width"],
-                    depth,
+                    brain.camera_resolutions.height,
```
Comment (chriselion) [inline gym-unity/gym_unity/envs/__init__.py]: Note: this is broken - should be brain.camera_resolutions[0].height, etc.

**Pull request merged**